### ISSUE TITLE
Nullable Reference Types and init properties

### ISSUE THREAD

**Issue by JanEggers** [dotnet/roslyn#49352]
current behavior:        

```csharp
public class Person
        {
            //this generates warning CS8618: Non-nullable property 'Name' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
            public string Name { get; init; }
        }

expected behavior:

//this should generate a warning
new Person { Name = null }

or
//this should generate a warning
Person p with { Name = null }

or new Person()
```

description in text: 

I use classes / records with init properties without ctor. currently all reference properties generate a warning if they dont have a default value in place.
I think that a bug. With init properties the warning should be generated where the object is constructed without providing a valid value.

I really like NRT's but its quite a lot of boilerplate code to initialize all the properties with default values that will be overridden anyway instead of leaving them blank and just force the code that constrcuts a new object to provide al required properties.

**Comment by huoyaoyuan** [comment]
This is by design *for now*. See dotnet/csharplang#3630 .

In short words, currently NRT only tracks assignment in constructors, because there are unpredictable counts of consumers, and there's no (concrete) contract to force initializing.

**Comment by JanEggers** [comment]
thx for pointing me to that issue
